js/components/graph/Sidebar.js

```diff
@@ -0,0 +1,178 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Focus from "./Focus";
+
+export default class Sidebar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      contentHidden: true,
+      focusDisabled: false,
+      graphActive: 0,
+      graphName: "",
+      toggled: false
+    }
+  }
+
+  componentWillUpdate(prevProps) {
```

Comment:
Getting warnings that `componentWillUpdate` is deprecated. Will this be a problem?